cylc/uiserver/schema.py

```diff
@@ -248,14 +254,205 @@ class Arguments:
+    
+    COUNT(*) AS n
+
+FROM
+    (SELECT
+        *,
+        NTILE (4) OVER (ORDER BY queue_time) queue_time_quartile,
+        NTILE (4) OVER (ORDER BY run_time) run_time_quartile,
+        NTILE (4) OVER (ORDER BY total_time) total_time_quartile
+        FROM
+            (SELECT
+                *,
+                STRFTIME('%s', time_run_exit) - STRFTIME('%s', time_submit) AS total_time,
+                STRFTIME('%s', time_run_exit) - STRFTIME('%s', time_run) AS run_time,
+                STRFTIME('%s', time_run) - STRFTIME('%s', time_submit) AS queue_time
+            FROM
+                task_jobs))
+WHERE
+    run_status = 0
+GROUP BY
+    name;
+            '''):
+                print('#', row)
+                jobs.append({
+                    'id': workflow.duplicate(
+                        cycle=row[1],
+                        task=row[0],
+                        job=row[2]
+                    ),
+                    'name': row[0],
+                    'cycle_point': row[1],
+                    'submit_num': row[2],
+                    'state': row[3],
+                    'started_time': row[4],
+                    'finished_time': row[5],
+                    'job_ID': row[6],
+                    'platform': row[7],
+                    'submitted_time': row[8],
+                    # Queue time stats
+                    'min_queue_time': row[9],
+                    'mean_queue_time': row[10],
+                    'max_queue_time': row[11],
+                    'std_dev_queue_time': (row[12] - row[10]**2)**0.5,
+                    'first_quartile_queue': row[13],
+                    'second_quartile_queue': row[14],
+                    'third_quartile_queue': row[15],
+                    # Run time stats
+                    'min_run_time': row[16],
+                    'mean_run_time': row[17],
+                    'max_run_time': row[18],
+                    'std_dev_run_time': (row[19] - row[17]**2)**0.5,
+                    'first_quartile_run': row[20],
+                    'second_quartile_run': row[21],
+                    'third_quartile_run': row[22],
+                    # Total
+                    'min_total_time': row[23],
+                    'mean_total_time': row[24],
+                    'max_total_time': row[25],
+                    'std_dev_total_time': (row[26] - row[24] ** 2) ** 0.5,
+                    'first_quartile_total': row[27],
+                    'second_quartile_total': row[28],
+                    'third_quartile_total': row[29],
+                    'count': row[30]
+                })
+    return jobs
+
+
+class UISTask(Task):
+
+    platform = graphene.String()
+    min_total_time = graphene.Int()
+    mean_total_time = graphene.Int()
+    max_total_time = graphene.Int()
+    std_dev_total_time = graphene.Float()
+    first_quartile_queue = graphene.Float()
+    second_quartile_queue = graphene.Float()
+    third_quartile_queue = graphene.Float()
```

Comment:
It may be worth considering combining the quartile fields e.g: `graphene.List(graphene.Float)` to cut down on the number of distinct fields.